### PR TITLE
8264293: Create implementation for NSAccessibilityMenu protocol peer

### PR DESCRIPTION
**Pull request by azuev-java**
Added implementation for all menu related protocol peers;
Native methods moved to CommonComponentAccessibility so they are called on correct peers;

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8264293](https://bugs.openjdk.java.net/browse/JDK-8264293): Create implementation for NSAccessibilityMenu protocol peer
 * [JDK-8264296](https://bugs.openjdk.java.net/browse/JDK-8264296): Create implementation for NSAccessibilityPopUpButton protocol peer
 * [JDK-8264295](https://bugs.openjdk.java.net/browse/JDK-8264295): Create implementation for NSAccessibilityMenuItem protocol peer
 * [JDK-8264294](https://bugs.openjdk.java.net/browse/JDK-8264294): Create implementation for NSAccessibilityMenuBar protocol peer

### Reviewers
 * [Pankaj Bansal](https://openjdk.java.net/census#pbansal) (@pankaj-bansal - **Reviewer**) ⚠️ Review applies to 264f8ee66169b2c83412c6c136530467679cb927
 * [Anton Tarasov](https://openjdk.java.net/census#ant) (@forantar - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6421/head:pull/6421` \
`$ git checkout pull/6421`

Update a local copy of the PR: \
`$ git checkout pull/6421` \
`$ git pull https://git.openjdk.java.net/jdk pull/6421/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6421`

View PR using the GUI difftool: \
`$ git pr show -t 6421`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6421.diff">https://git.openjdk.java.net/jdk/pull/6421.diff</a>

</details>

### CODE REVIEW

**Comment by victordyakov** [comment]
@pbansal please review

**Comment by victordyakov** [comment]
@forantar please review

**Review by pankaj-bansal — APPROVED** [review]
Looks good to me. Menu Accessibility works fine with these changes.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m [hunk 1]
```diff
@@ -152,6 +152,10 @@ + (void) initializeRolesMap {
     [rolesMap setObject:@"ListAccessibility" forKey:@"list"];
     [rolesMap setObject:@"OutlineAccessibility" forKey:@"tree"];
     [rolesMap setObject:@"TableAccessibility" forKey:@"table"];
+    [rolesMap setObject:@"MenuBarAccessibility" forKey:@"menubar"];
```
Comment (savoptik) [inline src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m:155]: Please increase the capacity of the rolesMap dictionary if you add a new role.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m [hunk 2]
```diff
@@ -1253,3 +1257,50 @@ - (BOOL)accessibilityPerformIncrement {
...
+ * Method:    menuOpened
+ * Signature: (I)V
+ */
+JNIEXPORT void JNICALL Java_sun_lwawt_macosx_CAccessible_menuOpened
+    (JNIEnv *env, jclass jklass, jlong element)
+{
+    JNI_COCOA_ENTER(env);
+    [ThreadUtilities performOnMainThread:@selector(postMenuOpened)
+        on:(CommonComponentAccessibility *)jlong_to_ptr(element)
+        withObject:nil waitUntilDone:NO];
```
Comment (forantar) [inline src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m:1272]: Please move `waitUntilDone` to the new line for consistency.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/MenuAccessibility.m [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#import "MenuAccessibility.h"
+
+/*
+ * Implementing a protocol that represents menus both as submenu and as a
+ * MenuBar components
+ */
+@implementation MenuAccessibility
+- (NSAccessibilityRole _Nonnull)accessibilityRole
+{
+        return [[[self parent] javaRole] isEqualToString:@"combobox"]
```
Comment (forantar) [inline src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/MenuAccessibility.m:35]: Please correct the indentation.

**Comment by forantar** [comment]
Besides what @savoptik mentioned, I left only cosmetics comments. Overall looks ok to me.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m (hunk 1 above)
Comment (azuev-java) [reply src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m:155]: Fixed.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/MenuAccessibility.m (hunk 3 above)
Comment (azuev-java) [reply src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/MenuAccessibility.m:35]: Fixed.

File: src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m (hunk 2 above)
Comment (azuev-java) [reply src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/CommonComponentAccessibility.m]: Fixed.

**Comment by azuev-java** [comment]
> Besides what @savoptik mentioned, I left only cosmetics comments. Overall looks ok to me.

Cosmetics are also important so i fixed them too.

**Comment by forantar** [comment]
Looks fine now.

**Review by forantar — APPROVED** [review]

**Pull request closed without merging**